lib/base_service.js

```diff
@@ -107,35 +107,50 @@ BaseService.prototype._getOptions = function(opts) {
             global: true
         }
     })
-    .command('docker-start', 'starts the service in a Docker container')
-    .command('docker-test', 'starts the test process in a Docker container')
-    .command('build', 'builds the service\'s package and deploy repo', function(yargs) {
-        return yargs.option({
-            f: {
-                alias: 'force',
-                default: false,
-                describe: 'force the operation to execute',
-                type: 'boolean'
-            },
-            d: {
-                alias: 'deploy-repo',
-                default: false,
-                describe: 'build only the deploy repo',
-                type: 'boolean'
-            },
-            s: {
-                alias: 'reshrinkwrap',
-                default: false,
-                describe: 'rebuild shrinkwrap.json by removing and regenerating after npm install',
-                type: 'boolean'
-            },
-            r: {
-                alias: 'review',
-                default: false,
-                describe: 'send the patch to Gerrit after building the repo',
-                type: 'boolean'
-            }
-        });
+    .command('docker-start', 'starts the service in a Docker container', {
+        nobuild: {
+            default: false,
+            describe: 'do not build nor execute the container, only generate the Dockerfile spec',
+            type: 'boolean'
+        }
+    })
+    .command('docker-test', 'starts the test process in a Docker container', {
+        nobuild: {
```

Comment:
The use case we have is to only get the DockerFile generated, though I guess one could also be interested in solely building the image and not running anything.   What about having top level commands such as:

 service-runner docker-file   # would generate the DockerFile
 service-runner docker-build  # generate DockerFile + docker build
 service-runner docker-start
 service-runner docker-test

Namely factors out the DockerFile generation and image building.